api/agent/pure_runner.go

```diff
@@ -626,11 +641,112 @@ DataLoop:
 	return state.waitError()
 }
 
+// Runs a status call using status image with baked in parameters.
+func (pr *pureRunner) runStatusCall(ctx context.Context, result *runner.RunnerStatus) {
+
+	log := common.Logger(ctx)
+	start := time.Now()
+
+	// construct call
+	var c models.Call
+
+	// Most of these arguments are baked in. We might want to make this
+	// more configurable.
+	c.ID = id.New().String()
+	c.Path = "/"
+	c.Image = pr.statusImageName
+	c.Type = "sync"
+	c.Format = "json"
+	c.TmpFsSize = 0
+	c.Memory = 0
+	c.CPUs = models.MilliCPUs(0)
+	c.URL = "/"
+	c.Method = "GET"
+	c.CreatedAt = common.DateTime(start)
+	c.Config = make(models.Config)
+	c.Config["FN_FORMAT"] = c.Format
+	c.Payload = "{}"
+
+	// Try to fetch ctx deadline
+	deadline, ok := ctx.Deadline()
+	if ok {
+		c.Timeout = int32(math.Min(float64(deadline.Sub(start)/time.Second), float64(1)))
+	} else {
+		c.Timeout = 10
+	}
+	c.IdleTimeout = 1
+
+	// TODO: reliably shutdown this container after executing one request.
```

Comment:
Is this reliable shutdown in scope for this PR?  If no, do we run any risk of having these status containers linger and behave in unexpected ways?

Reply:
Absence of 'execute-once' (aka maxRequests = 1) will make these tests more efficient since if status container is not idle for more than 1 seconds, it'll continue handling queries. We probably want to test a mixture of both cases, so as a follow up, I'd expect us to use a maxRequests=2 to test both fresh hot container *and* hot container handling subsequent queries.

Reply:
Or maxRequests=1 to keep things simple for time being. Until then, IdleTimeout=1 is best we can do here.